include/swift/AST/DiagnosticEngine.h

```diff
@@ -378,6 +378,77 @@ namespace swift {
       return fixItReplaceChars(Start, End, {});
     }
   };
+
+  /// \brief Class to track, map, and remap diagnostic severity and fatality
+  ///
+  class DiagnosticState {
+  public:
+    /// \brief Describes the current behavior to take with a diagnostic
+    enum class Behavior : uint8_t {
+      Unspecified,
+      Ignore,
+      Note,
+      Warn,
+      Err,
```

Comment:
Yes, I was going for some more verbs and also something that was visibly distinct from DiagnosticKind, but I don't think that's worth the confusion. I'll switch them back to Warning and Error. This might get refactored or rewritten with categories and exposing it higher anyways.